### ISSUE TITLE
Unwanted yaw or pitch from time to time, if it happens it's unstoppable

### ISSUE THREAD

**Issue by JanSeidlerPrivate** [Dimencia/DU-Orbital-Hud#55]
Sometimes, the script suddenly causes the ship to yaw or pitch. If that happens, it can't be stopped. Relogging, reapplying the script, nothing helps. I had the yawing four times, the pitching only once (but that one crashed me into the ground).

If I remove the script by applying "(Pilot) Flying Construct", the yawing/pitching stops immediately. Reactivating any of the ButtonHUD versions I have (3.88, 3.94, 3.96 and 3.97) cause the ship to yawn/pitch again. This happens regardless of whether I choose Virtual Joystick or Keyboard.

I noticed this problem can happen when I approach crowded areas which cause a lag spike. I don't know, maybe the script saves some values and tries to correct yaw/pitch then?

If it helps I can make a video of it or stream it to you guys on Discord.

**Comment by Archaegeo** [comment]
Im not saying its not occurring to you, but when its sometimes instead of
everytime, and not happening to everyone, and the fact the script
doesnt control flying unless you engage some control feature, are you sure
you didnt?

Also had you recently tabbed and clicked something else and your frames go
to 5-10fps when this happens?

More details definitely needed to help track a sometimes problem

**Comment by JanSeidlerPrivate** [comment]
I figured out the cause at least:

the problem fixed itself after I removed the databank from my ship that was connected to the ship. I had used ALT+7 multiple times to delete / save the data, and before removing the databank, I had receiver the message that the data has been deleted. Still, whenever I used the Command Seat and pressed G, the ship yawed right constantly. After removing the databank, the ship/script is behaving as expected. (No other script is running in this ship, no other data was in the databank.)

Maybe some old data doesn't get deleted by a new version of the script but somewhere the script is accessing the old data if it's available? Just guessing here, but I can definitely say your script is the only thing running in this ship.

As this issue is solved for me, you can absolutely close this, but if it happens to somebody else, please point them to remove the databank entirely from their ship. :)

**Comment by Archaegeo** [comment]
We were saving ship status, that might be doing it.  Dimencia can look into it.  The databank is super helpful to have.

**Comment by Dimencia** [comment]
Hopefully fixed, seems like a combo of the DU bug that starts you spinning randomly, and saving vars making it keep happening after exiting the seat.  Very hard to test if it's fixed, will leave this open for a little bit in case someone comes back with the same issue

**Comment by Archaegeo** [comment]
I am closing this as it seems fixed (3 days now)  If this comes back someone can open new issue.
